extensions/integration-tests/src/test/schemaCompare.test.ts

```diff
@@ -270,6 +274,231 @@ suite('Schema compare integration test suite @DacFx@', () => {
 		const excludeResult3 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, t2Difference, false, azdata.TaskExecutionMode.execute);
 		assertIncludeExcludeResult(excludeResult3, true, 0, 0);
 	});
+
+	test('Cancel schema compare dacpac to dacpac comparison', async function () {
+		this.timeout(5 * 60 * 1000);
+		assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+		const now = new Date();
+		const operationId = 'testOperationId_' + now.getTime().toString();
+
+		const source: mssql.SchemaCompareEndpointInfo = {
+			endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac1,
+			serverDisplayName: '',
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+			connectionDetails: undefined
+		};
+		const target: mssql.SchemaCompareEndpointInfo = {
+			endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac2,
+			serverDisplayName: '',
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+			connectionDetails: undefined
+		};
+
+		const promiseCompare = schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, null);
```

Comment:
How are you guaranteeing that it won't finish by the time schemaCompareCancel is called?

Reply:
This is based off of the time it takes for schema compare to finish. 
It is actually an "ideal" scenario where the user presses Compare button and then goes for Cancel right after that.

Reply:
I'm saying for purposes of the test here you're trying to test that cancelling the operation does what you expect. But the issue is that you're never guaranteeing that the compare won't just finish very quickly and thus have the cancel fail to go through, which means this is going to be a flakey test since sometimes it may finish quickly and fail the test while other times it may behave as you expect and pass just fine. 

Tests should be (need to be) deterministic to avoid these kinds of scenarios.

Reply:
I agree to that.
Is there any way that you can think of to make it more deterministic? await-ing on schema compare cancel and then awaiting on schemaCompare call, might help. But still that wouldn't make it 100% deterministic. 
Apart from the fact that compare does take time, I can't think of any way of making it a sure-shot test.

Reply:
if you set your stub that does compare in the service so that it waits for a really long time(longer than test timeout ) then returns the result then you can guarantee the test is actually canceling 

---
In reply to: [587980735](https://github.com/microsoft/azuredatastudio/pull/14479#discussion_r587980735) [](ancestors = 587980735)

Reply:
The problem with mocking the schema compare service by adding wait before the actual schemaCompare service call- can be that cancel may be called before the schemaCompare is called. Hence, it wouldn't be sure-shot either. 
Hence, I have removed this test, since it may become flaky down the line. And also because there are a couple of unit tests for testing the ADS side of cancel code.